cpp/CMakeLists.txt

```diff
@@ -685,6 +685,15 @@ set_source_files_properties(
   src/transform/transform.cpp
   PROPERTIES COMPILE_DEFINITIONS "_FILE_OFFSET_BITS=64"
 )
+# We need to hide symbols in the logger TU by default because those symbols include spdlog symbols
+# that are compiled in because we use it as a header-only library. This is problematic because other
+# libraries that also do the same thing will have their own copies, and they will all be weak
+# symbols so which one is used is undefined. Marking these symbols as hidden makes them local and
+# ensures that internal calls to spdlog functions correctly use the local copy.
+set_source_files_properties(
+  src/utilities/logger.cpp PROPERTIES COMPILE_OPTIONS
+                                      "-fvisibility=hidden;-fvisibility-inlines-hidden"
+)
```

Comment:
because we are transitively bringing in the header only version from rmm

Reply:
Thanks for the clarification, Rob. I added this info to the comment.

Reply:
I expect this PR to be merged before https://github.com/rapidsai/cudf/pull/15982, but once that is merged all this will be removed as that is the correct fix

Reply:
Yeah since libcudf only needs this one fix I'm fine pushing forward with the C++ wheel before we get around to fixing symbol visibility. That won't be an option for the rest of RAPIDS anyway.